start.js

```diff
@@ -13,25 +13,25 @@ function runServer() {
       if (body !== "") {
         body = JSON.parse(body);
       }
-      const parsedUrl = req.url.split("/");
-      if (/soldiers.*/.test(parsedUrl[1]) && parsedUrl.length < 4) {
+      const parsedUrl = (req.url.split("/"))[1];
+      if (/soldiers.*/.test(parsedUrl)) {
```

Comment:
Won't this accept both /soldiers/ and /soldiers123/ ? Why not just use simple comparison?

Reply:
If this is truly the case, and you found a bug, fix it the right way. Create a test which fails for this reason and then fix this.
Keep the test as part of your project's test of course.